src/ecdsa/test_ecdsa.py

```diff
@@ -33,23 +33,25 @@ def test_signature_validity(Msg, Qx, Qy, R, S, expected):
   def test_pk_recovery(Msg, R, S, Qx, Qy):
     
     sign = Signature(R,S)
-    pks = sign.recover_public_keys(digest_integer(Msg), curve_192, generator_192, generator_192.order())
+    pks = sign.recover_public_keys(digest_integer(Msg), generator_192)
 
     print_("Test pk recover")
 
     if len(pks) > 0:
 
       # Test if the signature is valid for all found public keys
       for pk in pks:
-        print_("Recovered pk: %s" % pk)
-        test_signature_validity(Msg, pk.x(), pk.y(), R, S, True)
+        q = pk.point
+        print_("Recovered q: %s" % q)
+        test_signature_validity(Msg, q.x(), q.y(), R, S, True)
 
       # Test if the original public key is in the set of found keys
-      original_pk = ellipticcurve.Point(curve_192, Qx, Qy)
-      if original_pk in pks:
-        print_("Original pk was found")
+      original_q = ellipticcurve.Point(curve_192, Qx, Qy)
+      points = map(lambda pk: pk.point ,pks)
```

Comment:
list comprehensions are considered more pythonic: `[pk.point for pk in pks]`

Reply:
Shure, I changed it